### PR TITLE
ci: Add Pre-release action to synchronize branches

### PR DESCRIPTION
**Pull request by alexs-mparticle**
## Instructions
 1. PR target branch should be against `development`
 2. PR title name should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-title-check.yml
 3. PR branch prefix should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-branch-check-name.yml

 ## Summary
 - This action synchronizes a release version of the web sdk between all necessary branches

 ## Testing Plan
 - [ ] Was this tested locally? If not, explain why.
 - N/A This is a CI/CD Update

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/SQDSDKS-5965

### CODE REVIEW

File: .github/workflows/synchronize-branches.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        - name: Checkout master branch
+          uses: actions/checkout@v3
+          with:
+            fetch-depth: 0
+            repository: ${{ github.repository }}
+            token: ${{ secrets.MP_SEMANTIC_RELEASE_BOT }}
+            ref: master
+
+        - name: Merge release branch into master branch
+          if: ${{ github.event.inputs.dryRun == 'false' }}
```
Comment (alexs-mparticle) [inline .github/workflows/synchronize-branches.yml:51]:
```suggestion
          if: ${{ github.event.inputs.dryRun == false }}
```

File: .github/workflows/synchronize-branches.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+          run: |
+            git pull origin release/${{ github.event.inputs.versionNumber}}
+        
+        # We will eventually replace master with main
+        - name: Push release commits to main and master
+          if: ${{ github.event.inputs.dryRun == 'false' }}
```
Comment (alexs-mparticle) [inline .github/workflows/synchronize-branches.yml:57]:
```suggestion
          if: ${{ github.event.inputs.dryRun == false }}
```

File: .github/workflows/synchronize-branches.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+          run: |
+                git push origin HEAD:master
+                git push origin HEAD:main
+
+        - name: Push release commits to development
+          if: ${{ github.event.inputs.dryRun == 'false' }}
```
Comment (alexs-mparticle) [inline .github/workflows/synchronize-branches.yml:63]:
```suggestion
          if: ${{ github.event.inputs.dryRun == false }}
```

File: .github/workflows/synchronize-branches.yml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+          run: |
+            git push origin HEAD:development
+
+        - name: Push release commits to chore/dependabot
+          if: ${{ github.event.inputs.dryRun }}
```
Comment (alexs-mparticle) [inline .github/workflows/synchronize-branches.yml:68]:
```suggestion
          if: ${{ github.event.inputs.dryRun == false }}
```

File: .github/workflows/synchronize-branches.yml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+          run: |
+            git push origin HEAD:chore/dependabot
+
+        - name: Push release commits to Release Order Branches
+          if: ${{ github.event.inputs.dryRun }}
```
Comment (alexs-mparticle) [inline .github/workflows/synchronize-branches.yml:73]:
```suggestion
          if: ${{ github.event.inputs.dryRun == false }}
```

**Review by rmi22186 — CHANGES_REQUESTED** [review]
Mostly nits, but looks great overall.

**Comment by rmi22186** [comment]
The commit name should also probably not be `Pre-release`, as it is the actual CDN release step.

**Review by rmi22186 — APPROVED** [review]

**Pull request merged**